CHANGES.md

```diff
@@ -1,5 +1,26 @@
 # Changes
 
+## edge-21.5.2
+
+This edge release updates the proxy-init container to check whether the iptables
+rules have already been added, which prevents errors if the proxy-init container
+is restarted. Also, the linkerd cli now has tab completion for kubernetes
+resources, saving you precious keystrokes! Finally, the proxy has been updated
+with several fixes and improvements.
+
+* Added instructions to `build.md` for using a locally built proxy
+  (thanks @jroper!)
+* Added support for kubernetes resource aware tab completion
+* Updated `proxy-init` to skip configuring firewall if rules exists
+* Fixed viz uninstall to delete all RBAC objects (thanks @aryan9600!)
+* Improved diagnostics for rejected profile discovery
+* Added proxy logic to set the l5d-client-id header
+* Reduced proxy resource usage when there are no profiles
+* Updated proxy to fail connections when HTTP detection fails
```

Comment:
Hm where is this note from? Is this supposed to be about when `l5d-dst-override` header is present? I would definitely mention include more detail about this note because the proxy doesn't fail when HTTP detection fails.

Reply:
This note and the one below come from the [notes](https://github.com/linkerd/linkerd2/pull/6107) for proxy v2.143.0

This specific note comes from [linkerd2-proxy/pull/979](https://github.com/linkerd/linkerd2-proxy/pull/979)

Reply:
Okay so this is specific to the admin server. I would include that level of detail because everywhere else HTTP detection can fail and things will proceed.